crates/common/consensus/src/electra/beacon_state.rs

```diff
@@ -151,9 +161,16 @@ pub struct BeaconState {
     pub eth1_deposit_index: u64,
 
     // Registry
+    #[cfg(not(feature = "zkvm"))]
     pub validators: VariableList<Validator, U1099511627776>,
+    #[cfg(feature = "zkvm")]
+    pub validators: VariableList<Validator, U536870912>,
```

Comment:
instead of doing it like this, could you pull out U1099511627776 and U536870912 define them as a type like `ValiatorLength` or whatever the constant is called in the spec, rather then defining optional fields. If you could apply this to the rest of the PR

```
    #[cfg(not(feature = "zkvm"))]
    type Hi = U32;
    #[cfg(feature = "zkvm")]
    type Hi = U16
    
    pub struct Car {
         pub validators: VariableList<Validator, Hi>,
    }
```

^ like this